src/main/java/com/intellij/internal/statistic/eventLog/validator/rules/utils/ValidationSimpleRuleFactory.java

```diff
@@ -187,6 +197,27 @@ private FUSRule createExpressionUtilRule(@NotNull List<String> nodes) {
     return nodes;
   }
 
+  private static int getPairBracket(String s, Integer index){
+    int curIndex;
+
+    Stack<Integer> opensStack = new Stack<>();
```

Comment:
You don't need a Stack here, because stored values are not used and we don't need synchronisation here. It could be replaced by int counter. Increment it when bracket opens and decrement when bracket closes.

Reply:
Thank you.